### PR TITLE
Fix pass by reference bug when applying changes to all installers

### PR DESCRIPTION
**Pull request by ryfu-msft**
This PR addresses a bug where List objects passed by reference were causing errors when serializing the manifest as seen in the image below (InstallerSuccessCodes produces weird characters):

MicrosoftTeams-image

The fix was to assign the object by value rather than by reference which requires creating new instances of the edited values

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-create/pull/164)

### CODE REVIEW

File: src/WingetCreateCLI/Commands/NewCommand.cs [hunk 1]
```diff
@@ -321,7 +322,55 @@ private static void ApplyChangesToIndividualInstallers(Installer installerCopy,
...
+                    {
+                        prop.SetValue(installer, fieldValue);
+                    }
+                    else if (fieldValue is IList)
+                    {
+                        Type elementType = prop.PropertyType.GetGenericArguments()[0];
+                        Type listType = typeof(List<>).MakeGenericType(elementType);
+                        var list = Activator.CreateInstance(listType, fieldValue);
+                        prop.SetValue(installer, list);
+                    }else if (fieldValue is Dependencies dependencies)
```
Comment (palenshus) [inline src/WingetCreateCLI/Commands/NewCommand.cs:336]: formatting

File: src/WingetCreateCLI/Commands/NewCommand.cs [hunk 2]
```diff
@@ -321,7 +322,55 @@ private static void ApplyChangesToIndividualInstallers(Installer installerCopy,
...
+                if (fieldValue is List<PackageDependencies> packageDependencies)
+                {
+                    List<PackageDependencies> packageDependenciesList = new List<PackageDependencies>();
+                    packageDependencies.ForEach(i => packageDependenciesList.Add(
+                        new PackageDependencies { PackageIdentifier = i.PackageIdentifier, MinimumVersion = i.MinimumVersion }));
+                    prop.SetValue(installer.Dependencies, packageDependenciesList);
+                }
+                else if (fieldValue is IList)
+                {
+                    Type elementType = prop.PropertyType.GetGenericArguments()[0];
```
Comment (palenshus) [inline src/WingetCreateCLI/Commands/NewCommand.cs:370]: we have this code twice, it should be a method

File: src/WingetCreateCLI/Commands/NewCommand.cs (hunk 1 above)
Comment (ryfu-msft) [reply src/WingetCreateCLI/Commands/NewCommand.cs]: Fixed

File: src/WingetCreateCLI/Commands/NewCommand.cs (hunk 2 above)
Comment (ryfu-msft) [reply src/WingetCreateCLI/Commands/NewCommand.cs]: No longer relevant with your new changes

**Review by palenshus — APPROVED** [review]
:shipit:

**Pull request merged**